src/MIDebugEngine/AD7.Impl/AD7Disassembly.cs

```diff
@@ -162,6 +176,28 @@ public int Read(uint dwInstructions, enum_DISASSEMBLY_STREAM_FIELDS dwFields, ou
                     }
                 }
 
+                if (!string.IsNullOrWhiteSpace(instruction.File))
+                {
+                    if ((dwFields & enum_DISASSEMBLY_STREAM_FIELDS.DSF_DOCUMENTURL) != 0)
+                    {
+                        prgDisassembly[iOp].dwFields |= enum_DISASSEMBLY_STREAM_FIELDS.DSF_DOCUMENTURL;
+                        prgDisassembly[iOp].bstrDocumentUrl = instruction.File;
+                    }
+
+                    if ((dwFields & enum_DISASSEMBLY_STREAM_FIELDS.DSF_POSITION) != 0)
+                    {
+                        prgDisassembly[iOp].dwFields |= enum_DISASSEMBLY_STREAM_FIELDS.DSF_POSITION;
+                        prgDisassembly[iOp].posBeg = new TEXT_POSITION()
+                        {
+                            dwLine = instruction.Line
```

Comment:
>dwLine = instruction.Line [](http://example.com/codeflow?start=28&length=25)

We might want to expand this range to cover more than just the exsact line that GDB gives us. Consider something like:

```
   printf("%d, %d, %d, %d\n",
     Func1(),
     Func2(),
     Func3(),
     Func4()
     );
```

If you only give back one line, then that is all that will show up in disassembly. So you would see:

```
   printf("%d, %d, %d, %d\n", // or worse mabe ');'
   mov ...
```

We probably instead want to remember the line of the previous instruction, and expand the range to fill in the space (with some limits to avoid dumping huge ranges).

Reply:
Still need to address expanding the range over multiple lines.

Reply:
This is what we get with the VC++ Debug Engine:
![image](https://user-images.githubusercontent.com/3953714/143141083-24163049-7c83-48f1-bc24-26d373f232a3.png)

What I have currently:
![image](https://user-images.githubusercontent.com/3953714/143141107-f93e842b-eec5-4283-aec7-57223ba4d210.png)